pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -423,9 +423,20 @@ func (c *Cluster) startDeployments(mons []*monConfig, requireAllInQuorum bool) e
 		if err != nil {
 			return fmt.Errorf("failed to create mon %s. %+v", mons[i].DaemonName, err)
 		}
+		// For the initial deployment (first creation) it's expected to not have all the monitors in quorum
+		// However, in an event of an update, it's crucial to proceed monitors by monitors
+		// At the end of the method we perform one last check where all the monitors must be in quorum
+		requireAllInQuorum := false
+		err = c.waitForMonsToJoin(mons, requireAllInQuorum)
+		if err != nil {
+			return fmt.Errorf("failed to check mon quorum %s. %+v", mons[i].DaemonName, err)
+		}
 	}
 
 	logger.Infof("mons created: %d", len(mons))
+	// Final verification that **all** mons are in quorum
+	// Do not proceed if one monitor is still syncing
+	requireAllInQuorum = true
```

Comment:
This change to requireAllInQuorum seems independent from the rest of the change. I also don't see why we are enforcing this quorum. It has worked very well to only require quorum instead of all mons to be running so that the rest of the orchestration won't be blocked. When a new mon is created it is expected to be in quorum before continuing, but otherwise we only should require majority quorum for the orchestration to continue.

Reply:
If we don't do this, during the upgrade, we won't be able to stop a new mon if all the mons are not in quorum.
If you have 3 mons, and 2 in quorum, you have quorum but you cannot start any of the 2. So forcing all of them to be in quorum allows us to pursue the upgrade.
This does not affect the initial bootstrap of monitors.

Reply:
How about if we only set `requireAllInQuorum` when the versions of the mons are different? Then if it isn't an upgrade we can still have the advantage of not blocking the orchestration if a mon is down.